src/cborencoder.c

```diff
@@ -196,34 +194,19 @@
  * Structure used to encode to CBOR.
  */
 
-#ifndef CBOR_NO_DFLT_WRITER
-/**
- * Initializes a CborEncoder structure \a encoder by pointing it to buffer \a
- * buffer of size \a size. The \a flags field is currently unused and must be
- * zero.
- */
-void cbor_encoder_init(CborEncoder *encoder, uint8_t *buffer, size_t size, int flags)
-{
-    cbor_buf_writer_init(&encoder->wr, buffer, size);
-
-    cbor_encoder_cust_writer_init(encoder, &encoder->wr.enc, flags);
-}
-#endif
-
 /**
  * Initializes a CborEncoder structure \a encoder by pointing it to buffer \a
  * buffer of size \a size. The \a flags field is currently unused and must be
  * zero.
  */
-void cbor_encoder_cust_writer_init(CborEncoder *encoder, struct cbor_encoder_writer *w, int flags)
+void cbor_encoder_init(CborEncoder *encoder, cbor_encoder_writer *writer, int flags)
 {
-    encoder->writer = w;
-    encoder->remaining = 2;
+    encoder->writer = writer;
+    encoder->added = 0;
     encoder->flags = flags;
 }
 
-
-#ifndef CBOR_NO_FLOATING_POINT
+#if FLOAT_SUPPORT
```

Comment:
This won't work with Zephyr. We should translate our `CONFIG_FLOAT` to this

Reply:
Yes, I just figured out that I have not tested all possible combinations of Kconfig variables and need to do that. I will update the review shortly.

Reply:
@de-nordic This is not fixed. Am I wrong?